internal/t3xf/t3xf.go

```diff
@@ -0,0 +1,244 @@
+// Package t3xf provides routines for decoding and loading t3xf encoded files.
+//
+// t3xf (TTCN-3 Executable Format) is a binary representation of input TTCN-3
+// source text, suitable for execution by a native application, its main purpose
+// expected to be the execution of test cases against a system under test.
+//
+// t3xf can be thought of as one possible binary dump of the abstract syntax
+// tree created by a TTCN-3 compiler after all semantic checking has been
+// performed. t3xf is conceived as a file format. No provision is made for
+// streaming or for editing. Transmission of whole files is not only possible
+// but will be required in systems that support testing distributed across a
+// number of host machines.
+//
+// tasm (T3xf Assembly) is an extension to t3xf allowing objects to be loaded
+// lazily and hence reducing the time for loading t3xf files.
+package t3xf
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+// File represents a t3xf file.
+type File struct {
+	Sections Sections
+	Tables   Tables
+
+	version int              // T3XF Version
+	tasm    bool             // true, if file is in TASM format
+	width   int              // Instruction width in bytes
+	order   binary.ByteOrder // Endianess of file
+}
+
+// ReadFile reads a file from path and returns a File struct. An error is returned if
+// the file could not be read or has an unknown or invalid format.
+func ReadFile(path string) (*File, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	size := 0
+	if fi, err := f.Stat(); err == nil {
+		// Check if size fits in int
+		if fi.Size() <= int64(int(^uint(0)>>1)) {
+			size = int(fi.Size())
```

Comment:
just for completeness: shouldn't the else case return an error?

Reply:
Size argument was used to preallocate slices. A size of zero means no pre-allocation. However, I remove this optimization, because it was unused.